src/core/mac/sub_mac.cpp

```diff
@@ -394,13 +394,13 @@ void SubMac::StartCsmaBackoff(void)
 
         if (ShouldHandleTransmitTargetTime())
         {
-            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) <
-                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay -
-                    kCcaSampleInterval - kCslTransmitTimeAhead - kRadioHeaderShrDuration)
+            uint32_t aheadTime      = kCcaSampleInterval + kCslTransmitTimeAhead + kRadioHeaderShrDuration;
+            Time     timerStartTime = Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - aheadTime;
+            Time     txStartTime    = timerStartTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay;
+
+            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) < txStartTime)
```

Comment:
Thanks for breaking this line and making it into multiple variables. Much easier to read and follow 😃

Reply:
How about something similar to this:
```c++
            static constexpr uint32_t kAheadTime = kCcaSampleInterval + kCslTransmitTimeAhead + kRadioHeaderShrDuration;

            Time     txStartTime = Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime);

            txStartTime += (mTransmitFrame.mInfo.mTxInfo.mTxDelay - kAheadTime);

            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) < txStartTime)
            {
                mTimer.FireAt(txStartTime);
            }
```